### PR TITLE
Feature/ddd

### PR DESCRIPTION
**Pull request by ajax-yurkovskyi-t**

### CODE REVIEW

File: iot-management-device/user/src/main/kotlin/com/example/iotmanagementdevice/user/infrastructure/rest/dto/response/UserResponseDto.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.example.iotmanagementdevice.user.infrastructure.rest.dto.response
+
+data class UserResponseDto(
+    val name: String?,
```
Comment (ajax-marchenko-s) [inline iot-management-device/user/src/main/kotlin/com/example/iotmanagementdevice/user/infrastructure/rest/dto/response/UserResponseDto.kt:4]: why nullable?

File: settings.gradle.kts [hunk 2]
```diff
@@ -8,5 +8,9 @@ include(
     "gateway",
     "grpc-api",
     "internal-api",
-    "iot-management-device"
+    "iot-management-device",
+    "iot-management-device:device",
+    "iot-management-device:user",
+    "iot-management-device:role",
+    "iot-management-device:migration"
```
Comment (ajax-marchenko-s) [inline settings.gradle.kts:15]: change to alphabetic order

File: iot-management-device/user/src/main/kotlin/com/example/iotmanagementdevice/user/infrastructure/configuration/MongoConfig.kt [hunk 3]
```diff
@@ -1,13 +1,11 @@
-package com.example.iotmanagementdevice.config
+package com.example.iotmanagementdevice.user.infrastructure.configuration
```
Comment (ajax-marchenko-s) [inline iot-management-device/user/src/main/kotlin/com/example/iotmanagementdevice/user/infrastructure/configuration/MongoConfig.kt:1]:
```suggestion
package com.example.iotmanagementdevice.user.infrastructure.mongo.configuration
```

**Review by ajax-marchenko-s — COMMENTED** [review]
Great work

File: iot-management-device/device/src/test/kotlin/com/example/iotmanagementdevice/device/application/mapper/DeviceUpdateEventMapperTest.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package com.example.iotmanagementdevice.device.application.mapper
```
Comment (ajax-gnatiuk-s) [inline iot-management-device/device/src/test/kotlin/com/example/iotmanagementdevice/device/application/mapper/DeviceUpdateEventMapperTest.kt:1]: is it correct package?

File: iot-management-device/device/src/test/kotlin/com/example/iotmanagementdevice/device/application/mapper/DeviceUpdateEventMapperTest.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+import com.example.iotmanagementdevice.device.DeviceFixture.createDevice
+import com.example.iotmanagementdevice.device.DeviceFixture.deviceUpdatedEvent
+import com.example.iotmanagementdevice.device.application.mapper.impl.DeviceUpdateEventMapperImpl
```
Comment (ajax-gnatiuk-s) [inline iot-management-device/device/src/test/kotlin/com/example/iotmanagementdevice/device/application/mapper/DeviceUpdateEventMapperTest.kt:5]: `application.mapper.impl` looks wierd to me. Seems, we should change the package

File: iot-management-device/role/src/main/kotlin/com/example/iotmanagementdevice/role/application/output/RoleRepositoryOutPort.kt [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package com.example.iotmanagementdevice.role.application.output
```
Comment (ajax-gnatiuk-s) [inline iot-management-device/role/src/main/kotlin/com/example/iotmanagementdevice/role/application/output/RoleRepositoryOutPort.kt:1]: `.port.` is missed in the package

File: iot-management-device/device/src/main/kotlin/com/example/iotmanagementdevice/device/application/mapper/DeviceUpdateEventMapper.kt [hunk 7]
```diff
@@ -0,0 +1,35 @@
+package com.example.iotmanagementdevice.device.application.mapper
```
Comment (ajax-gnatiuk-s) [inline iot-management-device/device/src/main/kotlin/com/example/iotmanagementdevice/device/application/mapper/DeviceUpdateEventMapper.kt:1]: why mapper in package `application.mapper`

**Review by ajax-gnatiuk-s — CHANGES_REQUESTED** [review]

File: iot-management-device/device/src/main/kotlin/com/example/iotmanagementdevice/device/application/mapper/DeviceUpdateEventMapper.kt (hunk 7 above)
Comment (ajax-yurkovskyi-t) [reply iot-management-device/device/src/main/kotlin/com/example/iotmanagementdevice/device/application/mapper/DeviceUpdateEventMapper.kt]: move it to infrastructure.kafka.mapper

**Pull request merged**